app/models/user.rb

```diff
@@ -1,12 +1,16 @@
 class User < ApplicationRecord
+  before_validation {email.downcase!}
+  before_destroy :destroy_ensure_admin
+  before_update :update_ensure_admin
+
   validates :name, presence: true, length: {maximum: 30}
   validates :email, presence: true, length: {maximum: 90}, format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i }
-  before_validation {email.downcase!}
   validates :password, presence: true, length: {minimum: 6}
   has_secure_password
+
   has_many :tasks, dependent: :delete_all
-  before_destroy :destroy_ensure_admin
-  before_update :update_ensure_admin
+  has_many :labels, dependent: :delete_all
+
   private
   def update_ensure_admin
     throw(:abort) if self.admin == false && User.where(admin: true).count == 1
```

Comment:
`count == 1`の部分は`one?`が利用できると思います。関連して`zero?`や`many?`などのメソッドもあります。
```suggestion
    throw(:abort) if self.admin == false && User.where(admin: true).one?
```